src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -2206,5 +2206,27 @@ private void prepareIntervalResults(Map<Timestamp, IntervalResults> dataResultsM
             throw new Exception(AnalyzerErrorConstants.APIErrors.UpdateRecommendationsAPI.METRIC_EXCEPTION + e.getMessage());
         }
     }
+
+    /**
+     * Filters out maxDateQuery and namespace metrics based on the experiment type
+     * @param metricProfile  Metric profile to be used
+     * @param maxDateQuery   maxDateQuery metric to be filtered out
```

Comment:
why maxDateQuery required to filter ?

Reply:
It seems that both require maxDateQuery, so is the filter still necessary?

Reply:
`maxDateQuery` for both the experiment types is already executed before iterating through the list of other metrics  to calculate `interval_start_time` and `interval_end_time` - https://github.com/kruize/autotune/blob/master/src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java#L2031

it's redundant to execute it twice with other metrics present hence filtering it out.